### PR TITLE
"Actions" workflow to generate documentation

### LINKED ISSUE #66: Automatically regenerate documentation on merge to trunk
An "Action" to run Sphinx on the documentation and post the result to "Pages".

### PR DESCRIPTION
**Pull request by MatthewHambley**
Although reasonably elaborate this workflow doesn't handle the issues around releases.

Closes #66

### CODE REVIEW

File: .github/workflows/sphinx.yml [hunk 1]
```diff
@@ -0,0 +1,120 @@
+# Inspired by https://github.com/rkdarst/sphinx-actions-test/blob/master/.github/workflows/sphinx-build.yml
+
+name: sphinx
+on: [push, pull_request]
```
Comment (matthewrmshin) [inline .github/workflows/sphinx.yml:4]: You may want to restrict this to run only on push and pull requests against master, and/or push to tags?

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#push

File: .github/workflows/sphinx.yml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        if: ${{ github.event_name == 'push' }}
+        run: |
+          touch gh-pages/.nojekyll
+
+      - name: Deploy
+        if: ${{ github.event_name == 'push' }}
+        run: |
+          set -x
+          git_hash=$(git -C stylist rev-parse --short "${{ github.sha }}")
+          git -C gh-pages -c "user.name=Sphinx Workflow" -c user.email=sphinx@example.com commit -a -m "Documentation from $git_hash"
```
Comment (matthewrmshin) [inline .github/workflows/sphinx.yml:119]: What are the purposes of the `-c` options?

**Review by matthewrmshin — COMMENTED** [review]
I have added a comment and asked a question. The change looks reasonable otherwise.

File: .github/workflows/sphinx.yml (hunk 2 above)
Comment (MatthewHambley) [reply .github/workflows/sphinx.yml:119]: It causes `git` to change to that directory before acting. It means you don't need to `cd <working copy>` in the script.

File: .github/workflows/sphinx.yml (hunk 1 above)
Comment (MatthewHambley) [reply .github/workflows/sphinx.yml]: That is a good idea. I have stripped everything right back so it no longer supports developer branches. Release tags will have to be added later but this is a good "minimal solution."

**Comment by MatthewHambley** [comment]
Developing and testing these workflows is always painful and the only way I can see to test the final deployment of the pages is to merge the change and hope.

**Pull request merged**